src/cyw43_btuart.c

```diff
@@ -0,0 +1,297 @@
+/*
```

Comment:
I think this should be called `cyw43_bthci.c` similar to https://github.com/micropython/micropython/blob/master/drivers/esp-hosted/esp_hosted_bthci.c.

Reply:
We need to distinguish this file from a BT HCI driver that uses a different transport, like the SPI/SDIO interface.  Hence the need to put "uart" in the filename.

How about `cyw43_bthci_uart.c`?

Reply:
Yes, `xx_bthci_uart.c` is good. The other drivers above should be renamed to match this.

Reply:
I've renamed them here to match the cyw43: https://github.com/micropython/micropython/pull/16833.

Reply:
I've renamed this to `cyw43_bthci_uart.c` in this PR.